modules/javafx.graphics/src/test/java/test/com/sun/javafx/iio/common/ImageToolsTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2014, 2015, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.iio.common;
+
+import com.sun.javafx.iio.common.ImageTools;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+import java.util.Random;
+
+public class ImageToolsTest {
+    @Test
+    public void testIfComputeDimensionsReturnsCorrectValuesWhenAspectRatioIsPreserved() {
+        for (int i = 2; i < 1000; i += 2) {

```

Comment:
Is there any need to have long loops ( 1000s of iterations ) for  this kind of test?

Reply:
I don't like this test case very much at all.  I'm having trouble thinking of one that would a) cover all edge cases, and b) does not use the same algorithm in the test itself.  Covering all edge cases is tough since you take 4 int parameters, making this a huge problem space, not just the usual of  covering a couple of branches.

If a test is sufficient that shows the old code breaking and the new code working, then I'll be happy to rewrite it.  As it is, I'm unsure how to make it cover all cases without a couple of loops.

Reply:
I have no problem with loops at all. But do we need thousands of iterations?

Reply:
I've reduced the number of iterations, and added some additional asserts that accurately assert the expected values for the dimension that didn't reach it's maximum size.